src/theory/datatypes/sygus_extension.cpp

```diff
@@ -1594,20 +1594,21 @@ void SygusExtension::check()
       {
         isExc = false;
         //debugging : ensure fairness was properly handled
-        if (options().datatypes.sygusFair == options::SygusFairMode::DT_SIZE)
+        if (TraceIsOn("sygus-sb"))
```

Comment:
This should not be guarded, the `AlwaysAssert` should be checked regardless of whether the trace is on.